pkg/metrics/wal/wal.go

```diff
@@ -646,6 +646,14 @@ func (a *appender) AppendExemplar(ref uint64, _ labels.Labels, e exemplar.Exempl
 		}
 	}
 
+	// Check for duplicate vs last stored exemplar for this series, and discard those.
+	// Otherwise, record the current exemplar as the latest
+	prevExemplar := a.w.series.getLatestExemplar(ref)
+	if prevExemplar != nil && exemplarsEqual(*prevExemplar, e) {
+		return 0, nil
```

Comment:
I think this needs to return `ref` to satisfy interface expectations

Reply:
The intention was to mirror what Prometheus is doing [here](https://github.com/prometheus/prometheus/blob/0df3489275ed4357b12d98e737290992e19915ce/tsdb/head_append.go#L341-L343).

The original interface [mentions](https://github.com/prometheus/prometheus/blob/fe06f16c116a109330cce0bcb61c2c6728ab7227/storage/interface.go#L217) something about cases where the (returned?) reference is 0 and not being used for caching, so I'll look around and see if there's a specific reason they're doing that or I'll change it to return ref.

Reply:
Ah, actually, that's fine then, but let's copy the comment so it's obvious why we're returning 0.